core/model/modx/processors/resource/trash/purge.class.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Empties the recycle bin.
+ *
+ * @return boolean
+ *
+ * @package    modx
+ * @subpackage processors.resource
+ */
+class modResourceTrashPurgeProcessor extends modProcessor {
+
+    /** @var modResource $resource */
+    public $resource;
+
+    /**
+     * @var int The id of the resource to be deleted.
+     */
+    public $id;
+
+
+    public function checkPermissions() {
+        return $this->modx->hasPermission( 'purge_deleted' );
+    }
+
+    public function getLanguageTopics() {
+        return array( 'resource' );
+    }
+
+    /**
+     * @return bool|null|string
+     */
+    public function initialize() {
+        $this->id = $this->getProperty( 'id', false );
+        if ( empty( $this->id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $this->resource = $this->modx->getObject( 'modResource', $this->id );
+        if ( empty( $this->resource ) ) return $this->modx->lexicon( 'resource_err_nfs', array( 'id' => $this->id ) );
+
+        /* validate resource can be deleted */
+        //if (!$this->resource->checkPolicy(array('save' => true, 'delete' => true))) {
+        //    return $this->modx->lexicon('permission_denied');
+        //}
+        return true;
+    }
+
+    public function process() {
+        /* get resources */
+        $id = $this->getProperty( 'id', false );
+        if ( empty( $id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $resources = $this->modx->getCollection( 'modResource', array(
+                'deleted' => true,
+                'id:IN' => explode(',',$id ))
+        );
+        $count = count( $resources );
+        $this->modx->log( 1, "Resources found: " . $count );
+
+        //$this->resource = $this->modx->getObject('modResource', $id);
+
+        // prepare for multiple purge at once
+        $ids = array();
+        /** @var modResource $resource */
+        foreach ( $resources as $resource ) {
+            $ids[] = $resource->get( 'id' );
+        }
+
+        $this->modx->invokeEvent( 'OnBeforeEmptyTrash', array(
+            'ids' => &$ids,
+            'resources' => &$resources,
+        ) );
+
+        reset( $resources );
+        $ids = array();
+        /** @var modResource $resource */
+        foreach ( $resources as $resource ) {
+            if ( !$resource->checkPolicy( 'delete' ) ) continue;
+
+            $resourceGroupResources = $resource->getMany( 'ResourceGroupResources' );
+            $templateVarResources = $resource->getMany( 'TemplateVarResources' );
+
+            /** @var modResourceGroupResource $resourceGroupResource */
+            foreach ( $resourceGroupResources as $resourceGroupResource ) {
+                $resourceGroupResource->remove();
+            }
+
+            /** @var modTemplateVarResource $templateVarResource */
+            foreach ( $templateVarResources as $templateVarResource ) {
+                $templateVarResource->remove();
+            }
```

Comment:
While this exists in current `modResourceEmptyRecycleBinProcessor`, i'm wondering if 

* that's required (schema relationships should handle this i guess)
* it should not occur after the `$resource->remove()` method (to make sure it went fine)

Reply:
@rtripault I am not sure this is handled automatically. Will test this. Regarding the order removal, you mean we should first remove the resource, and only if that was fine, then remove the template vars? would be the right order, but how do we get the template vars once the resource is deleted?